duckduckgo/Android
# Notification Experiments 2 and 3

Task/Issue URL: https://app.asana.com/0/361428290920652/1112318511572574 & https://app.asana.com/0/361428290920652/1114095864120161
Tech Design URL: N/A

**Description**:
Add notification experiments 2 and 3

**Steps to test this PR**:

**TEST 1: Privacy Notification Trackers: Launch**
1. In `VariantManager` comment out everything other than the `me` variant and in `NotificationScheduler` change the time unit `val timeUnit = TimeUnit.DAYS` to minutes.
1. Run a clean version of the app
1. Browse to cnn.com
1. Kill the app and note that after a minute you see a notification informing of number of trackers blocked
1. Tap the notification text and enture the app launches
1. Ensure the "mnot_s_pp_8_0" and "mnot_l_pp_8_0" pixels have fired with numbers representing the trackers and upgrades
1. Relaunch the app and ensure the notification does not show again

**TEST 2: Privacy Notification Upgrades: Dismiss**
1. As above, in `VariantManager` comment out everything other than the `me` variant and in `NotificationScheduler` change the time unit `val timeUnit = TimeUnit.DAYS` to minutes.
1. Run a clean version of the app
1. Browse to facebook.com twice (once will be below threshold)
1. Kill the app and note that after a minute you see a notification informing of number of upgrades
1. Dismiss it
1. Ensure the "mnot_s_pp_0_2" and "mnot_c_pp_0_2" pixels have fired with numbers representing the number of trackers and upgrades
1. Relaunch the app and ensure the notification does not show again

**TEST 3: Privacy Notification No browsing**
1. In `VariantManager` comment out everything other than the `me` variant and in `NotificationScheduler` change the time unit `val timeUnit = TimeUnit.DAYS` to seconds.
1. Run a clean version of the app
1. Kill the app and note that after a moment you see a generic notification as no trackers or upgrades have happened yet

**TEST 4: Privacy Notification Browsing AND Upgrades**
1. In `VariantManager` comment out everything other than the `me` variant and in `NotificationScheduler` change the time unit `val timeUnit = TimeUnit.DAYS` to minutes.
1. Run a clean version of the app
1. Browse to cnn.com and facebook.com twice
1. Kill the app and note that after a minute you see a notification informing of number of trackers blocked and upgrades

**TEST 5: Privacy Clear Notification: Launch**
1. In `VariantManager` comment out everything other than `sc` and in `NotificationScheduler` change the time unit `val timeUnit = TimeUnit.DAYS` to seconds.
1. Run a clean version of the app
1. Background the app, note that after a few seconds the clear data notification shows
1. Tap it to launch the clear data screen
1. Ensure the "mnot_s_cd" and "mnot_l_cd" pixels have fired
1. Relaunch the app and ensure it does not show again

**TEST 6: Privacy Clear Notification: Dismiss**
1. As above, in `VariantManager` comment out everything other than `sc` and in `NotificationScheduler` change the time unit `val timeUnit = TimeUnit.DAYS` to seconds.
1. Run a clean version of the app
1. Background the app, note that after a few seconds the clear data notification shows
1. Dismiss it
1. Ensure the "mnot_s_cd" and "mnot_c_cd" pixels have fired
1. Relaunch the app and ensure it does not show again

**TEST 7: Migration**
1. Install the develop branch and run the app
1. Install this branch and ensure that the migration succeeds and nothing goes boom

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
